### PR TITLE
Fix `snap_size` of Line2DWidget and add tests.

### PR DESCRIPTION
**Pull request by ericpre**
In the previous PR, I forgot to check the `snap_size`, this is now done! This fixes the bug mentioned by @magnunor in #1785.
If you don't have time to check this now, don't worry, it is not urgent, I just wanted to checked where does the bug (from #1785) comes from.

### CODE REVIEW

File: hyperspy/drawing/_widgets/line2d.py [hunk 1]
```diff
@@ -139,7 +143,6 @@ def _set_axes(self, axes):
         # _set_axes overwrites self._size so we back it up
         size = self._size
         super(Line2DWidget, self)._set_axes(axes)
-        self._pos = np.tile(self._pos, (2, 1))
```
Comment (francisco-dlp) [inline hyperspy/drawing/_widgets/line2d.py]: Why did you remove this line?

Comment (ericpre) [reply hyperspy/drawing/_widgets/line2d.py]: It seems that I forgot to push a commit (which is done now), where the `_pos` is backed up and restored just after the call of `_set_axes()` of the parent class. In this case, I would not expect to need to tile `_pos`.
I don't remember the reason for this, I may have to come back later to this. It was not obvious to me what was going on, I may have miss something.
